src/go/rpk/pkg/cli/topic/consume.go

```diff
@@ -221,6 +221,9 @@ func (c *consumer) consume(ctx context.Context) {
 						r.Key = decKey
 					}
 				}
+				if r.Value == nil {
+					r.Value = []byte("null")
```

Comment:
Misread the file originally. I don't think we want this, this hijacks the output _always_ for any null value.
I don't have a solid idea yet, but one idea is to add a formatting directive to print null or true if the value is null. I'm not sure yet.

Reply:
We can handle this case better if we change our Value to a `*string` type: https://gist.github.com/WillemKauf/459b2f7ece149a54a6b61f35d5706ec1

Now, our output is

```
{
  "topic": "topic",
  "key": "this_is_a_record_with_null_string",
  "value": "null",
  "timestamp": 1726000933223,
  "partition": 0,
  "offset": 3
}
{
  "topic": "topic",
  "key": "this_is_not_a_tombstone_record",
  "value": "",
  "timestamp": 1726000937808,
  "partition": 0,
  "offset": 4
}
{
  "topic": "topic",
  "key": "this_is_a_tombstone_record",
  "value": null,
  "timestamp": 1726000941565,
  "partition": 0,
  "offset": 5
}
```

Reply:
Unfortunately we would break `omitempty`, as can be seen from the output of `this_is_not_a_tombstone_record`. I'm starting to think printing empty values isn't the worst behaviour in the world, but it is a change.

Reply:
+1 I'm ok with this change. I can make a corresponding change to franz-go to support querying null-ness vs. emptiness -- `%v{is_null}` or something -- but for the json output here in rpk, this is a fine fix.

Reply:
Awesome! This change has been made in the last force push to this PR.